File: cpp/ros2/src/ngg/src/publisher_member_function.cpp
#include "publisher_member_function.hpp"

MinimalPublisher::MinimalPublisher()
: Node("minimal_publisher"), count_(0) {
  publisher_ = this->create_publisher<std_msgs::msg::String>("topic", 10);
  publish_input();
}

void MinimalPublisher::publish_input() {
  for (int i = 0; i < 5 ; i++) {
    auto message = std_msgs::msg::String();
    RCLCPP_INFO(this->get_logger(), "Input the number:");
    std::cin >> message.data;
    publisher_ -> publish(message);
  }
}

int main(int argc, char * argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MinimalPublisher>());
  rclcpp::shutdown();
  return 0;
}

File: python/ros2/src/ngg/ngg/publisher/publisher_member_function.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class MinimalPublisher(Node):
    def __init__(self):
        super().__init__('minimal_publisher')
        self.publisher_ = self.create_publisher(String,'topic', 10)
        # timer_period = 0.5
        # self.timer = self.create_timer(timer_period, self.timer_callback)
        # self.i = 0

    def timer_callback(self):
        msg = String()
        msg.data = f'Hello World: {self.i}'
        self.publisher_.publish(msg)
        self.get_logger().info(f'Publishing: {msg.data}')
        self.i += 1

    def publish_cli_input(self):
        while True:
            msg = String()
            msg.data = input('Input a number: ')
            self.publisher_.publish(msg)
            self.get_logger().info(f'Publishing: {msg.data}')



def main(args=None):
    rclpy.init(args=args)
    minimal_publisher = MinimalPublisher() # ノードの初期化
    minimal_publisher.publish_cli_input()
    rclpy.spin(minimal_publisher)
    minimal_publisher.destroy_node()
    rclpy.shutdown()

